packages/block-library/src/navigation/test/use-navigation-menu.js

```diff
@@ -0,0 +1,211 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect, createRegistry } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+
+/**
+ * Internal dependencies
+ */
+import useNavigationMenu from '../use-navigation-menu';
+
+function createRegistryWithStores() {
+	// Create a registry and register used stores.
+	const registry = createRegistry();
+	[ coreStore ].forEach( registry.register );
+
+	const navigationConfig = {
+		kind: 'postType',
+		name: 'wp_navigation',
+		baseURL: '/wp/v2/navigation',
+		rawAttributes: [ 'title', 'excerpt', 'content' ],
+	};
+	// Register post type entity.
+	registry.dispatch( coreStore ).addEntities( [ navigationConfig ] );
+	return registry;
+}
+
+jest.mock( '@wordpress/data/src/components/use-select', () => {
+	// This allows us to tweak the returned value on each test.
+	const mock = jest.fn();
+	return mock;
+} );
+
+function setRecords( registry, menus ) {
+	const dispatch = registry.dispatch( coreStore );
+	dispatch.startResolution( 'getEntityRecords', [
+		'postType',
+		'wp_navigation',
+		{ per_page: -1, status: 'publish' },
+	] );
+	dispatch.finishResolution( 'getEntityRecords', [
+		'postType',
+		'wp_navigation',
+		{ per_page: -1, status: 'publish' },
+	] );
+	dispatch.receiveEntityRecords( 'postType', 'wp_navigation', menus, {
+		per_page: -1,
+		status: 'publish',
+	} );
+}
+
+function setCreatePermission( registry, allowed ) {
+	const dispatch = registry.dispatch( coreStore );
+	dispatch.receiveUserPermission( 'create/navigation', allowed );
+	dispatch.startResolution( 'canUser', [ 'create', 'navigation' ] );
+	dispatch.finishResolution( 'canUser', [ 'create', 'navigation' ] );
+}
+
+function setUpdatePermission( registry, ref, allowed ) {
+	const dispatch = registry.dispatch( coreStore );
+	dispatch.receiveUserPermission( `update/navigation/${ ref }`, allowed );
+	dispatch.startResolution( 'canUser', [ 'update', 'navigation', ref ] );
+	dispatch.finishResolution( 'canUser', [ 'update', 'navigation', ref ] );
+}
+
+function setDeletePermission( registry, ref, allowed ) {
+	const dispatch = registry.dispatch( coreStore );
+	dispatch.receiveUserPermission( `delete/navigation/${ ref }`, allowed );
+	dispatch.startResolution( 'canUser', [ 'delete', 'navigation', ref ] );
+	dispatch.finishResolution( 'canUser', [ 'delete', 'navigation', ref ] );
+}
+
+describe( 'useNavigationMenus', () => {
+	const navigationMenu1 = { id: 1, title: 'Menu 1', status: 'publish' };
```

Comment:
Let's add a test to cover the situation where a menu is requested but it's status isn't `publish`. That way we can add coverage to the `selectExistingMenu` function which determines a menu's existence based on `status`.

Reply:
Done in 352f73f